### PR TITLE
nixos/systemd-boot: add efiBootOptionDescription option

### PR DESCRIPTION
**Pull request by kurtkilpela**
## Description of changes
By default, `bootctl` will use "Linux Boot Manager" as the boot option description. This is not ideal when dual-booting. In systemd 252, the `--efi-boot-option-description` flag was added to permit custom description text.

When `boot.loader.systemd-boot.efiBootOptionDescription` option is not set, the `--efi-boot-option-description` is not used and `bootctl` is left to use its default value.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix [hunk 1]
```diff
@@ -38,7 +38,9 @@ let
 
     configurationLimit = if cfg.configurationLimit == null then 0 else cfg.configurationLimit;
 
-    inherit (cfg) consoleMode graceful editor;
+    inherit (cfg) consoleMode graceful editor efiBootOptionDescription;
+
+    setEfiBootOptionDescription = cfg.efiBootOptionDescription != null;
```
Comment (JulienMalka) [inline nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:43]: Why not do this test on the python side?

Comment (kurtkilpela) [reply nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:43]: I wasn't sure of how to do that reliably. If `efiBootOptionDescription` is `null`, the substitution logic will not change the value. I could recreate the "@efiBootOptionDescription@" through string concatenation to do the comparison. It seemed a little odd so I did it this way.

File: maintainers/maintainer-list.nix
Comment (Frontear) [inline maintainers/maintainer-list.nix:1]: Separate this commit from rest of commits

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix [hunk 2]
```diff
@@ -317,6 +319,21 @@ in {
...
     };
 
+    efiBootOptionDescription = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = ''
+        Invoke `bootctl` with the `--efi-boot-option-description` option,
+        which specifies the boot entry name.
+
+        This option allows a user to distringuish between separate bootloaders
```
Comment (Frontear) [inline nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix:329]: distinguish

File: maintainers/maintainer-list.nix
Comment (kurtkilpela) [reply maintainers/maintainer-list.nix:1]: Thank you. Fixed.

File: nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix (hunk 2 above)
Comment (kurtkilpela) [reply nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix]: Fixed.

**Comment by kurtkilpela** [comment]
The commit message has been updated to align with the nixos guidelines.

**Pull request closed without merging**